integration/sawtooth_integration/tests/test_identity.py

```diff
@@ -0,0 +1,399 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------
+
+import enum
+import subprocess
+import unittest
+from uuid import uuid4
+
+import cbor
+
+from sawtooth_processor_test.message_factory import MessageFactory
+
+from sawtooth_integration.tests.integration_tools import RestClient
+from sawtooth_integration.tests.integration_tools import wait_for_rest_apis
+
+from sawtooth_sdk.protobuf import batch_pb2
+
+
+class TestTransactorPermissioning(unittest.TestCase):
+
+    def setUp(self):
+        self.REST_ENDPOINT = "rest-api:8080"
+        wait_for_rest_apis([self.REST_ENDPOINT])
+        self.alice = Transactor('alice', self.REST_ENDPOINT)
+        self.bob = Transactor('bob', self.REST_ENDPOINT)
+        self.carol = Transactor('carol', self.REST_ENDPOINT)
+        self.dave = Transactor('dave', self.REST_ENDPOINT)
+
+        self.chuck = Transactor('chuck', self.REST_ENDPOINT)
+        self.mallory = Transactor('mallory', self.REST_ENDPOINT)
+
+        self.walter = Transactor('walter', self.REST_ENDPOINT)
+
+    def test_transactor_permissioning(self):
+        """Test the transactor permissioning system using the Identity
+        transaction family.
+
+        Notes:
+            The test works from general to specific on the transactor
```

Comment:
Given that these are basically separate tests and all the users are created in setup, does it makes sense to break these into individual test functions, `test_transactor_broad_permissioning`, `test_transactor_batch_signer_persmissioning`, etc

Reply:
I had worried about the state of the validator at the end of one test affecting another test.